### PR TITLE
add user authentication

### PR DESCRIPTION
**Pull request by anthonyseu**
TODO:
- [x] sha256 for password
- [x] user fluxible fetchr to do authentication instead of express service endpoint
- [x] signout function
- [x] adjust signup/login/signout UI

Bug: 
- [x] Refresh page, authenticated user info is not present

### CODE REVIEW

**Comment by jacgrady1** [comment]
let me know if u have any question about fetchr.

**Comment by anthonyseu** [comment]
@jacgrady1 @quiz1991 please review

File: actions/UserActions.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+export default {
+    userSignup(context, payload, done) {
+        let params = {};
+        let body = payload;
+        let config = {};
+        context.service.create('UserService', params, body, config, function(err, user) {
+            if (err) {
+                console.log('service err: ', err);
+            }
+            // context.dispatch('USER_SIGNUP', user);
```
Comment (anthonyseu) [inline actions/UserActions.js]: Should we redirect to login or just use this to login user. 

Both behavior can be found on various website.

File: components/Login.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import UserActions from '../actions/UserActions';
+import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField';
+import FlatButton from 'material-ui/FlatButton';
+
+const sha256 = require('js-sha256');
+// TODO: needs to explore better way to apply css
+const userFormStyle = {
```
Comment (anthonyseu) [inline components/Login.js]: Not sure if this is the right way to add style in react world

File: components/Nav.js [hunk 3]
```diff
@@ -3,9 +3,33 @@ import { NavLink } from 'fluxible-router';
 import UserStore from '../stores/UserStore';
 import connectToStores from 'fluxible-addons-react/connectToStores';
 import Avatar from 'react-avatar';
+import UserActions from '../actions/UserActions';
+
+import UserActionMenu from './UserActionMenu';
+
+import IconButton from 'material-ui/IconButton';
```
Comment (anthonyseu) [inline components/Nav.js]: Not used

**Comment by jacgrady1** [comment]
on it

**Comment by jacgrady1** [comment]
👍

**Pull request merged**